salt/engines/__init__.py

```diff
@@ -81,33 +81,6 @@ def __init__(self, opts, fun, config, funcs, runners, proxy, **kwargs):
         self.runners = runners
         self.proxy = proxy
 
-    # __setstate__ and __getstate__ are only used on Windows.
-    # We do this so that __init__ will be invoked on Windows in the child
-    # process so that a register_after_fork() equivalent will work on Windows.
-    def __setstate__(self, state):
-        self.__init__(
-            state['opts'],
-            state['fun'],
-            state['config'],
-            state['funcs'],
-            state['runners'],
-            state['proxy'],
-            log_queue=state['log_queue'],
-            log_queue_level=state['log_queue_level']
-        )
-
-    def __getstate__(self):
-        return {
-            'opts': self.opts,
-            'fun': self.fun,
-            'config': self.config,
-            'funcs': self.funcs,
-            'runners': self.runners,
-            'proxy': self.proxy,
-            'log_queue': self.log_queue,
-            'log_queue_level': self.log_queue_level
-        }
-
```

Comment:
Not sure that I understand - how do all of these things end out getting called? They're quite different than what's in the other files.

Reply:
On windows, and windows alone, the `__new__` method will keep a copy of the `*args` and `**kwargs` used to instantiate the class.
Then, again, only on windows, the get and set state methods will use that information to re-instsntiate the class.

Reply:
[Here](https://github.com/saltstack/salt/pull/55793/files#diff-0eb908cc193e4bf8dd9f96567a3abfebR741)